tesseract-server/src/handlers/logic_layer/geoservice.rs

```diff
@@ -31,23 +32,16 @@ pub fn query_geoservice(
     geoservice_query: &GeoserviceQuery,
     geo_id: &str
 ) -> Result<Vec<GeoServiceResponseJson>, Error> {
-    let mut query_url: String;
+    let base_url = Url::parse(geoservice_url.clone()).unwrap();
 
-    if &geoservice_url[geoservice_url.len()-1..] == "/" {
-        match geoservice_query {
-            GeoserviceQuery::Neighbors => query_url = format!("{}neighbors/{}", geoservice_url, geo_id),
-            GeoserviceQuery::Children => query_url = format!("{}relations/children/{}", geoservice_url, geo_id),
-            GeoserviceQuery::Parents => query_url = format!("{}relations/parents/{}", geoservice_url, geo_id),
-            _ => return Err(format_err!("This type of geoservice query is not yet supported"))
-        }
-    } else {
-        match geoservice_query {
-            GeoserviceQuery::Neighbors => query_url = format!("{}/neighbors/{}", geoservice_url, geo_id),
-            GeoserviceQuery::Children => query_url = format!("{}/relations/children/{}", geoservice_url, geo_id),
-            GeoserviceQuery::Parents => query_url = format!("{}/relations/parents/{}", geoservice_url, geo_id),
-            _ => return Err(format_err!("This type of geoservice query is not yet supported"))
-        }
-    }
+    let join_str = match geoservice_query {
+        GeoserviceQuery::Neighbors => format!("{}neighbors/{}", geoservice_url, geo_id),
```

Comment:
geoservice_url is being passed in to this str, won't that make it repeat because you also join later?

Reply:
Good call! Somehow the join was able to fix that for me so I didn't even notice!